src/measure_theory/integral/layercake.lean

```diff
@@ -0,0 +1,258 @@
+  { suffices : ∃ (G : ℝ → ℝ), measurable G ∧ g =ᵐ[volume.restrict (Ioi (0 : ℝ))] G,
+    { rcases this with ⟨G₁, mble_G₁, g_eq_G₁⟩,
+      use (λ a, max 0 (G₁ a)),
+      refine ⟨measurable.max measurable_const mble_G₁, _, _⟩,
+      { filter_upwards [g_nn, g_eq_G₁],
+        intros a ga_nn ga_eq,
+        simp only [←ga_eq, ga_nn, max_eq_right], },
+      { exact λ a, by simp only [pi.zero_apply, le_max_iff, le_refl, true_or], }, },
+    suffices : ae_measurable g (volume.restrict (Ioi (0 : ℝ))), from this,
+    have Ioi_eq_Union : Ioi (0 : ℝ) = ⋃ (n : ℕ), Ioc 0 (n : ℝ),
+    { rw Union_Ioc_eq_Ioi_self_iff.mpr _,
+      exact λ x _, exists_nat_ge x, },
+    rw [Ioi_eq_Union, ae_measurable_Union_iff],
+    intros n,
+    by_cases n = 0,
+    { simp only [h, nat.cast_zero, Ioc_self,
+                 measure.restrict_empty, _root_.ae_measurable_zero_measure], },
+    { exact (g_intble n (nat.cast_pos.mpr (zero_lt_iff.mpr h))).1.1.ae_measurable, }, },
+  rcases ex_G with ⟨G, ⟨G_mble, g_eq_G, G_nn⟩⟩,
+  have g_eq_G_on : ∀ (t : ℝ), g =ᵐ[volume.restrict (Ioc 0 t)] G,
+  from λ t, ae_mono (measure.restrict_mono Ioc_subset_Ioi_self le_rfl) g_eq_G,
+  have G_nn' : ∀ t > 0, 0 ≤ G t, from λ t t_pos, G_nn t,
+  have G_intble : ∀ t > 0, interval_integrable G volume 0 t,
+  { intros t t_pos,
+    refine ⟨integrable_on.congr_fun' (g_intble t t_pos).1 (g_eq_G_on t), _⟩,
+    convert integrable_on_empty,
+    exact Ioc_eq_empty_of_le (show 0 ≤ t, by linarith), },
+  have eq₁ : ∫⁻ (t : ℝ) in Ioi 0, μ {a : α | t ≤ f a} * ennreal.of_real (g t)
+             = ∫⁻ (t : ℝ) in Ioi 0, μ {a : α | t ≤ f a} * ennreal.of_real (G t),
+  { apply lintegral_congr_ae,
+    filter_upwards [g_eq_G],
+    intros a ha,
+    rw [ha], },
+  have eq₂ : ∀ ω, ∫ (t : ℝ) in 0..f ω, g t = ∫ (t : ℝ) in 0..f ω, G t,
+  { intro ω,
+    have fω_nn : 0 ≤ f ω, by simpa only using f_nn ω,
+    rw [interval_integral.integral_of_le fω_nn, interval_integral.integral_of_le fω_nn],
+    exact integral_congr_ae (g_eq_G_on (f ω)), },
+  rw eq₁,
+  simp_rw eq₂,
+  exact layercake_of_measurable μ f_nn f_mble G_intble G_mble (λ t t_pos, G_nn t),
+end
+
+theorem layercake_one
+  (μ : measure α) [sigma_finite μ]
+  {f : α → ℝ} (f_nn : 0 ≤ f) (f_mble : measurable f) :
+  ∫⁻ ω, ennreal.of_real (f ω) ∂μ = ∫⁻ t in Ioi 0, (μ {a : α | t ≤ f a}) :=
+begin
+  set cst := λ (t : ℝ), (1 : ℝ) with def_cst,
+  have cst_intble : ∀ t > 0, interval_integrable cst volume 0 t,
+  from λ _ _, interval_integrable_const,
+  have key := layercake μ f_nn f_mble cst_intble (eventually_of_forall (λ t, zero_le_one)),
+  simp_rw [def_cst, ennreal.of_real_one, mul_one] at key,
+  rw ← key,
+  apply congr_arg,
+  funext ω,
+  simp only [interval_integral.integral_const, sub_zero, algebra.id.smul_eq_mul, mul_one],
+end
+
+-- TODO: The assumption `(p_large : 0 < p)` should suffice, but the assumptions of
+-- `interval_integral.interval_integrable_rpow` and `interval_integral.integral_rpow`
+-- are too stringent to apply here.
```

Comment:
Now that we have `interval_integrable_rpow'`, is this easy to fix?

Reply:
@kkytola did you investigate this?